### ISSUE TITLE
Quarto adds invalid symbol while rendering to PDF

### ISSUE THREAD

**Issue by bhattmaulik** [quarto-dev/quarto-cli#9894]
### Bug description

I have created a quarto document, and I want to render it to a PDF. But it doesn't render to PDF, unless and until I also add kbl function from kableExtra package. A small reproducible example is as follows:

### Steps to reproduce

````
library(readxl)
library(knitr)
library(kableExtra)
#This will not work
df <- readxl::read_excel(readxl::readxl_example("datasets.xlsx"),.name_repair = "unique")
head(df)
#This will work
df <- readxl::read_excel(readxl::readxl_example("datasets.xlsx"),.name_repair = "unique")
head(df) |> kableExtra::kbl()
````

### Expected behavior

The document should render to PDF even without using kbl function.

### Actual behavior

It gives error

```
ERROR: 
compilation failed- error
Text line contains an invalid character.
l.210 ^^[
         [38;5;246m# A tibble: 6 x 5^^[[39m 

see document1.log for more information.
```

### Your environment

IDE: RStudio 2024.07.0 Build 192
OS: Windows 11

### Quarto check output

````
$ quarto
$ quarto check
Quarto 1.5.42
[>] Checking versions of quarto
 binary dependencies...
      Pandoc version 3.2.0: OK
      Dart Sass version 1.70.0:
 OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[>] Checking versions of quarto
 dependencies......OK
[>] Checking Quarto installatio
n......OK
      Version: 1.5.42
      Path: C:\Users\Hp\AppData
\Local\Programs\Quarto\bin
      CodePage: 1252

[>] Checking tools.............
.......OK
      TinyTeX: (external instal
l)
      Chromium: (not installed)

(|) Checking LaTeX.............
.......
(/) Checking LaTeX.............
.......
(-) Checking LaTeX.............
.......
(\) Checking LaTeX.............
.......
(|) Checking LaTeX.............
.......
(/) Checking LaTeX.............
.......
(-) Checking LaTeX.............
.......
(\) Checking LaTeX.............
.......
(|) Checking LaTeX.............
.......
(/) Checking LaTeX.............
.......
(-) Checking LaTeX.............
.......
(\) Checking LaTeX.............
.......
(|) Checking LaTeX.............
.......
(/) Checking LaTeX.............
.......
(-) Checking LaTeX.............
.......
(\) Checking LaTeX.............
.......
(|) Checking LaTeX.............
.......
(/) Checking LaTeX.............
.......
(-) Checking LaTeX.............
.......
(\) Checking LaTeX.............
.......
(|) Checking LaTeX.............
.......
(/) Checking LaTeX.............
.......
(-) Checking LaTeX.............
.......
(\) Checking LaTeX.............
.......
(|) Checking LaTeX.............
.......
(/) Checking LaTeX.............
.......
(-) Checking LaTeX.............
.......
(\) Checking LaTeX.............
.......
(|) Checking LaTeX.............
.......
(/) Checking LaTeX.............
.......
(-) Checking LaTeX.............
.......
(\) Checking LaTeX.............
.......
(|) Checking LaTeX.............
.......
(/) Checking LaTeX.............
.......
(-) Checking LaTeX.............
.......
(\) Checking LaTeX.............
[>] Checking LaTeX.............
.......OK
      Using: TinyTex
      Path: C:\Users\Hp\AppData
\Roaming\TinyTeX\bin\windows\
      Version: 2024

(|) Checking basic markdown ren
der....
(/) Checking basic markdown ren
der....
(-) Checking basic markdown ren
der....
(\) Checking basic markdown ren
der....
(|) Checking basic markdown ren
der....
(/) Checking basic markdown ren
der....
(-) Checking basic markdown ren
der....
(\) Checking basic markdown ren
der....
(|) Checking basic markdown ren
der....
(/) Checking basic markdown ren
der....
(-) Checking basic markdown ren
[>] Checking basic markdown ren
der....OK

[>] Checking Python 3 installat
ion....(None)

      Unable to locate an insta
lled version of Python 3.
      Install Python 3 from htt
ps://www.python.org/downloads/

(|) Checking R installation....
.......
(/) Checking R installation....
.......
(-) Checking R installation....
.......
(\) Checking R installation....
.......
[>] Checking R installation....
.......OK
      Version: 4.4.0
      Path: C:/PROGRA~1/R/R-44~
1.0
      LibPaths:
        - C:/Users/Hp/AppData/L
ocal/R/win-library/4.4
        - C:/Program Files/R/R-
4.4.0/library
      knitr: 1.47
      rmarkdown: 2.27

(|) Checking Knitr engine rende
r......
(/) Checking Knitr engine rende
r......
(-) Checking Knitr engine rende
r......
(\) Checking Knitr engine rende
r......
(|) Checking Knitr engine rende
r......
(/) Checking Knitr engine rende
r......
(-) Checking Knitr engine rende
r......
(\) Checking Knitr engine rende
r......
(|) Checking Knitr engine rende
r......
(/) Checking Knitr engine rende
r......
(-) Checking Knitr engine rende
r......
(\) Checking Knitr engine rende
r......
(|) Checking Knitr engine rende
r......
(/) Checking Knitr engine rende
r......
(-) Checking Knitr engine rende
r......
(\) Checking Knitr engine rende
r......
[>] Checking Knitr engine rende
r......OK
````

**Comment by mcanouil** [comment]
Could you share a **small self-contained "working" (reproducible)** example to work with, _i.e._, a complete Quarto document or a Git repository? The goal is to make it as easy as possible for us  to recreate your problem so that we can fix it: please help us help you! Thanks.

Here you only shared R code.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Comment by bhattmaulik** [comment]
Please find a MWE below. In the code above, I had forgotten to add the YAML header.

````
---
title: "Extra Character"
format: pdf
editor: visual
---

## Table with Extra Character

I have observed a weird thing that it adds some extra characters in a table. Because of this extra character, I can't render it to PDF. There are two workarounds to solve this:

1.  Change the output to `html`, or

2.  Add a call to `kbl` function from `kableExtra` package.

```{r}
#| label: tbl-not-working
#| tbl-cap: "This does not work"
library(readxl)
library(knitr)
library(kableExtra)
df <- readxl::read_excel(readxl::readxl_example("datasets.xlsx"),.name_repair = "unique")
head(df)
```

Now I am adding `kbl` function from `kableExtra` package.

```{r}
#| label: tbl-working
#| tbl-cap: "This works"
df <- readxl::read_excel(readxl::readxl_example("datasets.xlsx"),.name_repair = "unique")
head(df) |> 
  kbl()
````

**Comment by cderv** [comment]
> l.210 ^^[
>         [38;5;246m# A tibble: 6 x 5^^[[39m

These are ANSI character that are printed by tibble. Quarto does not have a way yet to handle ANSI and convert to the right format or clean the output. 

I am surprised this happens because  I thought ANSI printing should have been disabled when rendering a document and only used in R console. 

This should happen on the R package stack. I'll have a look to confirm.

**Comment by cderv** [comment]
So I confirm this is R package and not quarto. This is all about **crayon** and **cli** used throughout the packages from the tidyverse. 

Within **knitr** context, it should have been correctly opt out, but it seems it is not. 

You can do `options(cli.num_colors = 1L)`

```markdown
---
title: "Extra Character"
format: pdf
keep-md: true
keep-tex: true
---

```{r}
#| include: false
options(cli.num_colors = 1L)
```

## Table with Extra Character

I have observed a weird thing that it adds some extra characters in a table. Because of this extra character, I can't render it to PDF. There are two workarounds to solve this:

1.  Change the output to `html`, or

2.  Add a call to `kbl` function from `kableExtra` package.

```{r}
#| label: tbl-not-working
#| tbl-cap: "This does not work"
library(readxl)
library(knitr)
library(kableExtra)
df <- readxl::read_excel(readxl::readxl_example("datasets.xlsx"), 
                         .name_repair = "unique")
head(df)
```


````

You'll see correct results 
![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/fc579e06-10e0-44fe-87ea-fa870b5ce542)

You can also use an env ver `R_CLI_NUM_COLORS = 1L`


I'll try to see why this fails here - probably an issue in one of the package

**Comment by cderv** [comment]
This is opened in RStudio repo. We can follow there. 

Thanks again for the report!

**Comment by bhattmaulik** [comment]
Thank you for getting to the root of the problem (as always), and explaining it to me in the simplest terms. I am following the issue there now.

**Comment by cderv** [comment]
Update: This should be solved now started in 2024.07.0-daily+303
